velox/exec/tests/AggregateSpillBenchmarkBase.cpp

```diff
@@ -26,35 +26,117 @@ using namespace facebook::velox::exec;
 
 namespace facebook::velox::exec::test {
 
+namespace {
+std::unique_ptr<RowContainer> makeRowContainer(
+    const std::vector<TypePtr>& keyTypes,
+    const std::vector<TypePtr>& dependentTypes,
+    std::shared_ptr<velox::memory::MemoryPool>& pool) {
+  auto container = std::make_unique<RowContainer>(
+      keyTypes,
+      true, // nullableKeys
+      std::vector<Accumulator>{},
+      dependentTypes,
+      false, // hasNext
+      false, // isJoinBuild
+      false, // hasProbedFlag
+      false, // hasNormalizedKey
+      pool.get());
+  return container;
+}
+
+std::unique_ptr<RowContainer> setupSpillContainer(
+    const RowTypePtr& rowType,
+    uint32_t numKeys,
+    std::shared_ptr<velox::memory::MemoryPool>& pool) {
+  const auto& childTypes = rowType->children();
+  std::vector<TypePtr> keys(childTypes.begin(), childTypes.begin() + numKeys);
+  std::vector<TypePtr> dependents;
+  if (numKeys < childTypes.size()) {
+    dependents.insert(
+        dependents.end(), childTypes.begin() + numKeys, childTypes.end());
+  }
+  return makeRowContainer(keys, dependents, pool);
+}
+} // namespace
+
 void AggregateSpillBenchmarkBase::setUp() {
   SpillerBenchmarkBase::setUp();
 
-  rowContainer_ =
-      setupSpillContainer(rowType_, FLAGS_spiller_benchmark_num_key_columns);
-  spiller_ = std::make_unique<Spiller>(
-      exec::Spiller::Type::kAggregateInput,
-      rowContainer_.get(),
-      [&](folly::Range<char**> rows) { rowContainer_->eraseRows(rows); },
-      rowType_,
-      HashBitRange{29, 29},
-      rowContainer_->keyTypes().size(),
-      std::vector<CompareFlags>{},
-      fmt::format("{}/{}", spillDir_, FLAGS_spiller_benchmark_name),
-      FLAGS_spiller_benchmark_max_spill_file_size,
-      FLAGS_spiller_benchmark_write_buffer_size,
-      FLAGS_spiller_benchmark_min_spill_run_size,
-      stringToCompressionKind(FLAGS_spiller_benchmark_compression_kind),
-      memory::spillMemoryPool(),
-      executor_.get());
-  writeSpillData();
+  inputPool_ = rootPool_->addLeafChild("inputSpillerPool");
+  outputPool_ = rootPool_->addLeafChild("outputSpillerPool");
+
+  inputSpillerDir_ =
+      fmt::format("{}/{}/input", spillDir_, FLAGS_spiller_benchmark_name);
+  outputSpillerDir_ =
+      fmt::format("{}/{}/output", spillDir_, FLAGS_spiller_benchmark_name);
+  fs_->mkdir(inputSpillerDir_);
+  fs_->mkdir(outputSpillerDir_);
+
+  inputRowContainer_ = setupSpillContainer(
+      rowType_, FLAGS_spiller_benchmark_num_key_columns, pool_);
+  writeSpillData(inputRowContainer_);
+  outputRowContainer_ = setupSpillContainer(
+      rowType_, FLAGS_spiller_benchmark_num_key_columns, pool_);
+  writeSpillData(outputRowContainer_);
+
+  inputSpiller_ = makeSpiller(Spiller::Type::kAggregateInput);
```

Comment:
I think we could just use one spiller and memory pool and control the test procedure based on spiller type? The spiller type could be as a member of AggregateSpillBenchmarkBase test object? Thanks!

Reply:
Got it, thanks. I've updated it as you suggested by adding a type field and for loop the types of aggregate spiller. Should we make the aggregate spiller type a config parameter so that we can control the benchmark parameters for different types of aggregate spiller?